lua/orgmode/org/hyperlinks/init.lua

```diff
@@ -181,10 +181,10 @@ function Hyperlinks.get_link_to_headline(headline, path)
     id = headline:id_get_or_create()
   end
 
-  if not config.org_id_link_to_org_use_id or not id then
-    return ('file:%s::*%s'):format(path, title)
+  if config.org_id_link_to_org_use_id and id then
+    return ('id:%s::*%s'):format(id, title)
```

Comment:
Using the same url syntax as in file links enables the Url class to parse the parts correctly without adding logic.